src/SoftRobots/component/constraint/model/CableModel.inl

```diff
@@ -212,15 +249,34 @@ void CableModel<DataTypes>::internalInit()
 {
     initActuatedPoints();
     checkIndicesRegardingState();
+
+    if (d_method.getValue() == "geodesic" || d_method.getValue() == "sphere")
+        initCableActionAreas();
 }
 
 
 template<class DataTypes>
 void CableModel<DataTypes>::initActuatedPoints()
-{
+{   
     int nbIndices = d_indices.getValue().size();
     ReadAccessor<Data<VecCoord>> positions = m_state->readPositions();
     const SetIndexArray &indices = d_indices.getValue();
+    
+    ReadAccessor<Data<VecCoord>> centers = d_centers;
+    unsigned int nbCenters = centers.size();
+    if (nbCenters != 0)
+    {
+        if (nbIndices != 0)
+            msg_warning() <<"Both centers and indices are provided. Centers are used by default";
+
+        SetIndexArray &list = (*d_indices.beginEdit());
```

Comment:
It is not recommended to use the `beginEdit`/`endEdit` function.
Prefer `sofa::helper::getWriteOnlyAccessor`